File: BalancedBinaryTree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    
    public boolean Balanced(TreeNode root,int Ai[]) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        if(root==null){
            Ai[0]=0;
            return true;
        }
        int A[]=new int[1];
        int B[]=new int[1];
        if(!Balanced(root.left,A)) return false;
        if(!Balanced(root.right,B)) return false;
        Ai[0]=Math.max(A[0],B[0])+1;
        return Math.abs(A[0]-B[0])<=1;
    }
    public boolean isBalanced(TreeNode root){
        int A[]=new int[1];
        return Balanced(root,A);
    }
}

File: BalancedBinaryTree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool Balanced(TreeNode* root,int &c) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        if(!root){
            c=0;
            return true;
        }
        int a,b;
        if(!Balanced(root->left,a)) return false;
        if(!Balanced(root->right,b)) return false;
        c=max(a,b)+1;
        return abs(a-b)<=1;
    }
    bool isBalanced(TreeNode* root){
        int a;
        return Balanced(root,a);
    }
};
